### PR TITLE
ReleaseTest-DocComments

### PR DESCRIPTION
**Pull request by MarcusDenker**
This PR adds DocCommentsReleaseTest, a test class that creates a test for each doc comment (see #buildSuite on the class side).

- for now just test instance side methods (DrTest does that, we need to fix it there first and fix the broken cases)

This adds 892 tests

### CODE REVIEW

File: src/PharoDocComment-Tests/DocCommentsReleaseTest.class.st [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+	#category : #'PharoDocComment-Tests'
+}
+
+{ #category : #'building suites' }
+DocCommentsReleaseTest class >> buildSuite [
+		| suite methods |
+	suite := TestSuite named: 'Test Generated From Comments'.
+	"we do not run examples in tests (as they are used just for testing and might be broken by
+	design"
+	methods := Smalltalk globals allMethods reject: [ :method | method methodClass isTestCase ].
```
Comment (Ducasse) [inline src/PharoDocComment-Tests/DocCommentsReleaseTest.class.st]: can you remove the Smalltalk globals ref?

File: src/PharoDocComment-Tests/DocCommentsReleaseTest.class.st [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+"
+Class {
+	#name : #DocCommentsReleaseTest,
+	#superclass : #TestCase,
+	#category : #'PharoDocComment-Tests'
+}
+
+{ #category : #'building suites' }
+DocCommentsReleaseTest class >> buildSuite [
+		| suite methods |
```
Comment (Ducasse) [inline src/PharoDocComment-Tests/DocCommentsReleaseTest.class.st]: tab too much

File: src/PharoDocComment-Tests/DocCommentsReleaseTest.class.st (hunk 1 above)
Comment (MarcusDenker) [reply src/PharoDocComment-Tests/DocCommentsReleaseTest.class.st]: yes. But it shows that we are missing a mechanism: the only thing I can do is to use the environment of the test class. But that is not what I mean: I want to write "System, give me all methods, regardless of the environment that this class is in". 

Of course "Smalltalk globals" is not good: for one it is a global reference. And it hardcodes "give me all methods from the global environment", which is not the same as "give me all methods, not only those from classes in the global environment"

Right now there is only one environment, so "self class environment" and "Smalltalk globals" happen to point to the same environment.

I will use self class environment for now.

**Comment by Ducasse** [comment]
Super nice.
testNoUnusedTemporaryVariablesLeft – MacOSX64.ReleaseTests.CleanCode.NoUnusedVariablesLeftTest
testNoShadowedVariablesInMethods – MacOSX64.ReleaseTests.Release.ReleaseTest

**Comment by MarcusDenker** [comment]
failing tests are timeouts on the mac machine, it sometimes happens. I think it is running slow due to virtualisation.

**Comment by MarcusDenker** [comment]
Ah, I think it makes sense to add a reset for the ASTCache, this way the cached doc comments node instances get cleaned up.
(I have an idea how to speed up parsing, then we will not need to cache them anymore, but that is for Pharo10)

**Pull request merged**